processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -210,22 +212,36 @@ public AggregatorFactory getMergingFactory(AggregatorFactory other) throws Aggre
   public abstract List<String> requiredFields();
 
   /**
-   * Get the type name of the intermediate type for this aggregator. This is the same as the type returned by
+   * Get the "intermediate" {@link ValueType} for this aggregator. This is the same as the type returned by
    * {@link #deserialize} and the type accepted by {@link #combine}. However, it is *not* necessarily the same type
    * returned by {@link #finalizeComputation}.
+   */
+  public abstract ValueType getType();
+
+  /**
+   * Get the type for the final form of this this aggregator, i.e. the type of the value returned by
+   * {@link #finalizeComputation}. This may be the same as or different than the types expected in {@link #deserialize}
+   * and {@link #combine}.
+   * @return
+   */
+  public ValueType getFinalizedType()
+  {
+    return getType();
```

Comment:
IMO, it would be better for this to be abstract, since this makes it too easy for people to forget to override it.

Reply:
changed to abstract